api/app/services/llm_extraction/unified_batch_processor.py

```
@@ -0,0 +1,786 @@
+            # Cache result
+            self._add_to_cache(cache_key, result)
+
+            logger.info(
+                f"Unified batch processing: {len(all_questions)} user questions from {len(messages)} total messages "
+                f"in {len(conversations_data)} conversations (processing_time={result.processing_time_ms}ms)"
+            )
+
+            return result
+
+        except json.JSONDecodeError as e:
+            logger.error(f"Failed to parse LLM response as JSON: {e}")
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)